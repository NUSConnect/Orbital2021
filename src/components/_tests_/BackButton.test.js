import * as React from 'react'
import BackButton from '../BackButton'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

it('renders correctly', () => {
  const tree = renderer.create(<BackButton />)
  expect(tree).toMatchSnapshot()
})

it('on press button', () => {
  const mockGoBackFn = jest.fn()
  const { getByTestId } = render(<BackButton goBack={mockGoBackFn} />)
  fireEvent.press(getByTestId('back'))
  expect(mockGoBackFn).toHaveBeenCalled()
})
