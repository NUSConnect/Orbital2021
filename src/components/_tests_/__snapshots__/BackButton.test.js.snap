// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "left": 4,
      "opacity": 1,
      "position": "absolute",
      "top": 30,
    }
  }
  testID="back"
>
  <Image
    source={1}
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
